server/core/region.go

```diff
@@ -749,58 +749,52 @@ func (r *RegionsInfo) GetStoreLearnerCount(storeID uint64) int {
 
 // RandPendingRegion randomly gets a store's region with a pending peer.
 func (r *RegionsInfo) RandPendingRegion(storeID uint64, ranges []KeyRange) *RegionInfo {
-	startKey, endKey := r.GetKeys(ranges)
+	startKey, endKey := GetKeys(ranges)
```

Comment:
Why not bypass `ranges` to the tree? Randomly pick a range may have problem: if some ranges have no region and the range is randomly selected, the selection process will end with no region selected.

Reply:
> Why not bypass ranges to the tree?

Can you explain more?

Reply:
something like
```
sum := 0
result := nil
for _, r := range ranges {
  count := tree.Count(r)
  sum += count
  if rand.Intn(sum) < count {
    result = tree.Rand(r)
  }
}
return result
```